src/mlpack/methods/ann/ffn.hpp

```diff
@@ -142,6 +142,16 @@ class FFN
   void Predict(arma::mat& predictors, arma::mat& results);
 
   /**
+   * Predict the responses to a given set of predictors. The responses will
+   * reflect the output of the given output layer as returned by the
+   * output layer function.
+   *
+   * @param predictors Input predictors.
+   * @return Output predictions of responses into.
+   */
+  arma::mat Predict(arma::mat predictors);
+
```

Comment:
Hi @zoq , I suspect this fix will lead to a forever copy. 
`this->predictors = std::move(predictors);` 
After this fix the RHS will be `const mat&&`, then the `=` operator will call `=(const mat&)` rather than `=(mat&&)`, which I think will lead to a copy.
See cpp.sh/2itol and [this](https://stackoverflow.com/questions/38917130/does-stdmove-works-with-lvalue-references-how-stdmove-works-on-standard-con)